packages/pluggableWidgets/gallery-web/src/components/Gallery.tsx

```diff
@@ -0,0 +1,58 @@
+import { createElement, ReactElement, ReactNode } from "react";
+import classNames from "classnames";
+
+export interface GalleryProps<T> {
+    className?: string;
+    desktopItems: number;
+    items: T[];
+    itemRenderer: (
+        renderWrapper: (children: ReactNode, className?: string, onClick?: () => void) => ReactElement,
+        item: T
+    ) => ReactNode;
+    phoneItems: number;
+    tabletItems: number;
+    tabIndex?: number;
+}
+
+export function Gallery<T>(props: GalleryProps<T>): ReactElement {
+    return (
+        <div
+            className={classNames(
+                "widget-gallery",
+                props.className,
+                `widget-gallery-lg-${props.desktopItems}`,
+                `widget-gallery-md-${props.tabletItems}`,
+                `widget-gallery-sm-${props.phoneItems}`
+            )}
+            data-focusindex={props.tabIndex || 0}
+        >
+            {props.items.map((item: T, index: number) =>
+                props.itemRenderer((children, className, onClick) => {
+                    return (
+                        <div
+                            key={`item_${index}`}
```

Comment:
Can we use not use an identifier from the item instead of the index here? Think that would make more sense and be more future proof in case for example we want drag and drop

Reply:
fixed